modules/ingester/ingester_test.go

```diff
@@ -339,9 +341,73 @@ func TestFlush(t *testing.T) {
 	}
 }
 
+func TestDedicatedColumns(t *testing.T) {
+	tmpDir, err := os.MkdirTemp("/tmp", "")
+	require.NoError(t, err, "unexpected error getting tempdir")
+	defer os.RemoveAll(tmpDir)
+
+	limits := overrides.Limits{}
+	flagext.DefaultValues(&limits)
+	limits.DedicatedColumns = []backend.DedicatedColumn{{Scope: "span", Name: "foo", Type: "string"}}
+
+	i := defaultIngesterWithOverrides(t, tmpDir, limits)
+	inst, _ := i.getOrCreateInstance("test")
+	require.NotNil(t, inst)
+
+	dec := model.MustNewSegmentDecoder(model.CurrentEncoding)
+
+	// create some search data
+	id := make([]byte, 16)
+	_, err = rand.Read(id)
+	require.NoError(t, err)
+	trace := test.MakeTrace(10, id)
+	b1, err := dec.PrepareForWrite(trace, 0, 0)
+	require.NoError(t, err)
+
+	// push to instance
+	require.NoError(t, inst.PushBytes(context.Background(), id, b1))
+
+	// Write wal
+	require.NoError(t, inst.CutCompleteTraces(0, true))
+
+	assert.Equal(t, limits.DedicatedColumns, inst.headBlock.BlockMeta().DedicatedColumns)
+
+	// TODO: This search should find a match once the read path is supported
+	ctx := user.InjectOrgID(context.Background(), "test")
+	searchReq := &tempopb.SearchRequest{Query: "{span.foo=\"bar\"}"}
+	results, err := inst.Search(ctx, searchReq)
+	require.NoError(t, err)
+	assert.Len(t, results.Traces, 0)
+
+	blockID, err := inst.CutBlockIfReady(0, 0, true)
+	require.NoError(t, err)
+
+	// TODO: This check should be included as part of the read path
+	inst.blocksMtx.RLock()
+	for _, b := range inst.completingBlocks {
+		assert.Equal(t, limits.DedicatedColumns, b.BlockMeta().DedicatedColumns)
+	}
+	inst.blocksMtx.RUnlock()
+
+	// Complete block
+	err = inst.CompleteBlock(blockID)
+	require.NoError(t, err)
+
+	// TODO: This check should be included as part of the read path
+	inst.blocksMtx.RLock()
+	for _, b := range inst.completeBlocks {
+		assert.Equal(t, limits.DedicatedColumns, b.BlockMeta().DedicatedColumns)
+	}
+	inst.blocksMtx.RUnlock()
+}
+
 func defaultIngesterModule(t testing.TB, tmpDir string) *Ingester {
+	return defaultIngesterWithOverrides(t, tmpDir, defaultLimitsTestConfig())
```

Comment:
`nitpick` This could be an opportunity to rename `defaultLimitsTestConfig` to `defaultOverridesTestConfig` or just `defaultOverrides`.